src/json.js

```diff
@@ -137,6 +135,61 @@ export function tryParseJson(json, opt_onFailed) {
   }
 }
 
+/**
+ * Recursively checks strict equality of items in nested arrays and objects.
+ *
+ * @param {JSONValueDef} a
+ * @param {JSONValueDef} b
+ * @param {number} depth The maximum recursion depth. Must be finite.
+ * @return {boolean}
+ * @throws {Error} If depth argument is not finite.
+ */
+export function recursiveEquals(a, b, depth = 3) {
+  if (!isFinite(depth)) {
+    throw new Error('depth arg must be finite: ' + depth);
+  }
+  if (a === b) {
+    return true;
+  }
+  if (a === null || b === null) {
+    return false;
+  }
+  if (typeof a !== typeof b) {
+    return false;
+  }
+  if (depth <= 0) {
+    return false;
+  }
+  if (Array.isArray(a) && Array.isArray(b)) {
+    if (a.length !== b.length) {
+      return false;
+    }
+    for (let i = 0; i < a.length; i++) {
+      if (!recursiveEquals(a[i], b[i], depth - 1)) {
+        return false;
+      }
+    }
+    return true;
+  }
+  if (typeof a === 'object' && typeof b === 'object') {
```

Comment:
JSONValueDef currently has type `*` so we could actually get functions and Symbols here. ☹️